fleetshard/pkg/central/client/client.go

```diff
@@ -49,56 +48,122 @@ func NewCentralClient(central private.ManagedCentral, address, pass string) *Cli
 	}
 }
 
-// SendRequestToCentral ...
-func (c *Client) SendRequestToCentral(ctx context.Context, requestBody interface{}, path string) (*http.Response, error) {
-	jsonBytes, err := json.Marshal(requestBody)
-	if err != nil {
-		return nil, errors.Wrap(err, "marshalling new request to central")
+// NewCentralClientNoAuth creates a new client for central without authentication.
+func NewCentralClientNoAuth(central private.ManagedCentral, address string) *Client {
+	return &Client{
+		central: central,
+		address: address,
+		httpClient: http.Client{
+			Transport: insecureTransport,
+		},
 	}
-	req, err := http.NewRequest(http.MethodPost, c.address+path, bytes.NewReader(jsonBytes))
+}
+
+// SendRequestToCentral sends the request message to central and returns the http response.
+func (c *Client) SendRequestToCentral(ctx context.Context, requestMessage proto.Message, method, path string) (*http.Response, error) {
+	req, err := c.createRequest(ctx, requestMessage, method, path)
 	if err != nil {
 		return nil, errors.Wrap(err, "creating HTTP request to central")
 	}
-	req.SetBasicAuth("admin", c.pass)
-	req = req.WithContext(ctx)
+	if c.pass != "" {
+		req.SetBasicAuth("admin", c.pass)
+	}
 	resp, err := c.httpClient.Do(req)
 	if err != nil {
 		return nil, errors.Wrap(err, "sending new request to central")
 	}
 	return resp, nil
 }
 
-// SendGroupRequest ...
+func (c *Client) createRequest(ctx context.Context, requestMessage proto.Message, method, path string) (*http.Request, error) {
+	var body io.Reader
+	if requestMessage != nil {
+		marshaller := jsonpb.Marshaler{
+			EmitDefaults: true,
+			Indent:       "  ",
+		}
+		jsonString, err := marshaller.MarshalToString(requestMessage)
+		if err != nil {
+			return nil, errors.Wrap(err, "marshalling new request to central")
+		}
+		body = strings.NewReader(jsonString)
```

Comment:
```suggestion
		jsonBytes, err := marshaller.Marshal(requestMessage)
		if err != nil {
			return nil, errors.Wrap(err, "marshalling new request to central")
		}
		body = bytes.NewReader(jsonBytes)
```
no need for string here